cores/epoxy/Arduino.cpp

```diff
@@ -20,16 +20,27 @@
 // -----------------------------------------------------------------------
 // Arduino methods emulated in Unix
 // -----------------------------------------------------------------------
+static uint32_t arduinoPins = 0;
 
 void yield() {
   usleep(1000); // prevents program from consuming 100% CPU
 }
 
 void pinMode(uint8_t /*pin*/, uint8_t /*mode*/) {}
 
-void digitalWrite(uint8_t /*pin*/, uint8_t /*val*/) {}
+void digitalWrite(uint8_t pin, uint8_t val) {
+  if( val == HIGH ){
+    arduinoPins |= ( 1LU << ( pin - 1 ) );
+  }
+  if( ( val == LOW ) && ( digitalRead( pin ) ) ){
+    arduinoPins ^= ( 1LU << ( pin - 1 ) );
+  }
+}
+
+int digitalRead(uint8_t pin) { 
+  return ( arduinoPins & ( 1LU << ( pin - 1 ) ) ) == ( 1LU << ( pin - 1 ) ); 
```

Comment:
Since we are testing for only a single bit, we can simplify this to:
```C++
return (arduinoPins & (1LU << pin) != 0;
```
This generated fewer machine instructions because it doesn't have to test against an extra argument.